src/edu.umn.cs.melt.copper.compiletime/src/main/java/edu/umn/cs/melt/copper/compiletime/auxiliary/counterexample/CounterexampleSearch.java

```diff
@@ -0,0 +1,601 @@
+package edu.umn.cs.melt.copper.compiletime.auxiliary.counterexample;
+
+import edu.umn.cs.melt.copper.compiletime.lrdfa.LR0DFA;
+import edu.umn.cs.melt.copper.compiletime.lrdfa.LR0ItemSet;
+import edu.umn.cs.melt.copper.compiletime.lrdfa.LRLookaheadSets;
+import edu.umn.cs.melt.copper.compiletime.parsetable.LRParseTableConflict;
+import edu.umn.cs.melt.copper.compiletime.spec.numeric.ContextSets;
+import edu.umn.cs.melt.copper.compiletime.spec.numeric.PSSymbolTable;
+import edu.umn.cs.melt.copper.compiletime.spec.numeric.ParserSpec;
+
+import java.util.*;
+
+/**
+ * The business logic for finding counter examples for parsing conflicts.
+ * Reading "Finding Counterexamples from Parsing Conflicts" by Isradisaikul & Myers 
+ * is highly recommended before attempting to edit this code.
+ * This is a more or less direct implementation of the process discussed in that paper.
+ * For the sake of complexity, this implements only non-unifying counterexamples.
+ * @author Kelton OBrien
+ */
+public class CounterexampleSearch {
```

Comment:
I'm not going to try to review the logic here.  But nothing looks obviously wrong to me.

Reply:
There are several debugging lines in here that print to `System.out`, and should be commented out.